### PR TITLE
must-gather gather-sysinfo unit tests

### PR DESCRIPTION
**Pull request by marioferh**
Add unit test of gather-sysinfo

gather-sysinfo is must-gather main tool

Signed-off-by: Mario Fernandez <mariofer@redhat.com>

### CODE REVIEW

File: tools/gather-sysinfo/gather-sysinfo_test.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	"/host/proc/sys/kernel/sched_domain/cpu*/domain*/flags",
+	"/host/sys/devices/system/cpu/offline",
+	"/host/sys/class/dmi/id/bios*",
+	"/host/sys/class/dmi/id/product_family",
+	"/host/sys/class/dmi/id/product_name",
+	"/host/sys/class/dmi/id/product_sku",
+	"/host/sys/class/dmi/id/product_version",
+}
+
+var dupEntries = []string{"/host/proc/cmdline"}
```
Comment (yanirq) [inline tools/gather-sysinfo/gather-sysinfo_test.go:29]: can we make the var names more clear ?

File: tools/gather-sysinfo/gather-sysinfo_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package main
```
Comment (fromanirh) [inline tools/gather-sysinfo/gather-sysinfo_suite_test.go:1]: I'm not a fan of using ginkgo for unit tests. We add a great amount of baggage with unclear benefits. What are the benefits of using ginkgo here?

File: tools/gather-sysinfo/gather-sysinfo_test.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
+package main
```
Comment (fromanirh) [inline tools/gather-sysinfo/gather-sysinfo_test.go:1]: we may want to strip the logic from the main source and move them into an **internal** package to make unit test easier

File: tools/gather-sysinfo/gather-sysinfo_test.go (hunk 1 above)
Comment (marioferh) [reply tools/gather-sysinfo/gather-sysinfo_test.go:29]: right

File: tools/gather-sysinfo/gather-sysinfo_suite_test.go (hunk 2 above)
Comment (marioferh) [reply tools/gather-sysinfo/gather-sysinfo_suite_test.go:1]: Same opinion here, but we decided to use ginkgo for everything.

File: tools/gather-sysinfo/gather-sysinfo_test.go (hunk 3 above)
Comment (fromanirh) [reply tools/gather-sysinfo/gather-sysinfo_test.go:1]: `tools/gather-sysinfo/$WHATEVER` can be a nice start

Comment (marioferh) [reply tools/gather-sysinfo/gather-sysinfo_test.go:1]: ok, thanks

**Comment by marioferh** [comment]
@yanirq please override gpc and operator upgrade. Thanks

**Review by fromanirh — COMMENTED** [review]
these are not unit tests for must gather, but unit tests for the main tool which we use in must-gather, `gather-sysinfo`. So this should be reflected in the commit message (which should also be less terse).

I still believe we should split the logic from the `gather-sysinfo` main package to make the tests easier and nicer, but I won't push too much for that.

So, conditional LGTM, need to have the commit message fixed and then we can move forward.

**Comment by marioferh** [comment]
> these are not unit tests for must gather, but unit tests for the main tool which we use in must-gather, `gather-sysinfo`. So this should be reflected in the commit message (which should also be less terse).
> 
> I still believe we should split the logic from the `gather-sysinfo` main package to make the tests easier and nicer, but I won't push too much for that.
> 
> So, conditional LGTM, need to have the commit message fixed and then we can move forward.

done

**Comment by marioferh** [comment]
thanks @fromanirh next step: https://github.com/openshift-kni/performance-addon-operators/issues/897

**Comment by marioferh** [comment]
@fromanirh @yanirq needs override

**Pull request merged**